packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -1184,10 +1184,15 @@ class LocalizationsGenerator {
     );
 
     final List<LocaleInfo> allLocales = _allBundles.locales.toList()..sort();
-    final String fileName = outputFileName.split('.')[0];
+    final List<String> fileParts = outputFileName.split('.');
```

Comment:
I think this could be done more simply:
```dart
  final int extensionIndex = outputFilename.indexOf('.');
  // TBD error checking and reporting:
  // - expect a non-empty filename (not ".foo.dart" - extensionIndex == 0)
  // - expect to find a final ".dart" suffix (not "foo" or "foo.dart.bar")
  final String fileName = outputFilename.substring(0, extensionIndex);
  final String fileExtension = outputFilename.substring(extensionIndex + 1);
  ```
  
  It may not be necessary to check the integrity of outputFilename ("error checking and reporting"), I didn't check its origins.

Reply:
Not sure if it's intentional, but there's a test that specifies an output file without ".dart" suffix:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/general.shard/generate_localizations_test.dart#L754

Reply:
I think it's OK to assume that an output file with no suffix is a mistake and to fix the case[s] in generate_localizations_test.dart that used such an output file.